src/setup.erl

```diff
@@ -30,15 +30,29 @@ require_node_count(undefined) ->
 require_node_count(_) ->
     ok.
 
-error_bind_address() ->
-    throw({error, "Cluster setup requires bind_addres != 127.0.0.1"}).
-
-require_bind_address("127.0.0.1", undefined) ->
-    error_bind_address();
-require_bind_address("127.0.0.1", <<"127.0.0.1">>) ->
-    error_bind_address();
-require_bind_address(_, _) ->
-    ok.
+error_local_bind_address() ->
+    throw({error, "Cluster setup requires a remote bind_address (not 127.0.0.1 nor ::1)"}).
+
+require_remote_bind_address(OldBindAddress, NewBindAddress) ->
+    case {OldBindAddress, NewBindAddress} of
+        {"127.0.0.1", undefined} -> error_local_bind_address();
+        {_, <<"127.0.0.1">>} -> error_local_bind_address();
+        {"::1", undefined} -> error_local_bind_address();
+        {_, <<"::1">>} -> error_local_bind_address();
+        {_, undefined} -> ok;
+        {_, PresentNewBindAddress} -> require_valid_bind_address(PresentNewBindAddress)
+    end.
```

Comment:
🤔 While this definitely works and I’m not requesting a change here, I’d like to take the opportunity to explain the pattern that is used on the right.

Erlang allows for function signature overloading. I.e. the same function can have different function bodies depending on the arguments you pass in. That’s what’s used in the old version of the code. In your version, you use a `case` construct that does a functionally equivalent thing. Each matching condition(the thing before the `->` could be a function argument, and the clause (the thing after the `->` could be the function body.

That is a little more idiomatic Erlang. It doesn’t make much of a difference for this code, but imagine a longer function with multiple nested case statements, you end up with a lot of indentation real quickly, while the function matching pattern keeps the indentation to a minimum and the code more readable.